news/feature-4432.md

```diff
@@ -0,0 +1,4 @@
+Make `persist_state_alloc_string` reuse the same entry if possible.
+
+The `persist_state_alloc_string` function has been modified to first call `persist_state_lookup_entry` and check if an allocation with the same name already exists. If it does and also the allocation is large enough to hold the target message then instead of creating a new allocation the existing one is used.
```

Comment:
Does this change have a user visible consequence? If so that should be documented in the news file. If no, the we might not need a news entry.

Reply:
No, this change has no user visible consequence.